modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -2993,6 +2994,64 @@ public void testDeleteObjectEntry() throws Exception {
 			publishedObjectDefinition);
 	}
 
+	@Test
+	public void testDeleteObjectEntryWithObjectDefinitionTree()
+		throws Exception {
+
+		Tree objectDefinitionTree = TreeTestUtil.createObjectDefinitionTree(
+			_objectDefinitionLocalService, _objectRelationshipLocalService,
+			true,
+			LinkedHashMapBuilder.put(
+				"A", new String[] {"AA", "AB"}
+			).put(
+				"AA", new String[] {"AAA", "AAB"}
+			).put(
+				"AB", new String[0]
+			).put(
+				"AAA", new String[0]
+			).put(
+				"AAB", new String[0]
+			).build());
+
+		ObjectDefinition rootObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionTree.getRootNode(
+				).getPrimaryKey());
+
+		Tree objectEntryTree = TreeTestUtil.createObjectEntryTree(
+			"1", _objectDefinitionLocalService, _objectEntryLocalService,
+			_objectFieldLocalService, _objectRelationshipLocalService,
+			rootObjectDefinition.getObjectDefinitionId());
+
+		ObjectEntry rootObjectEntry = _objectEntryLocalService.getObjectEntry(
+			objectEntryTree.getRootNode(
+			).getPrimaryKey());
+
+		// When users deletes a root object entry its child entries are also
+		// deleted
+
+		_objectEntryLocalService.deleteObjectEntry(
+			rootObjectEntry.getObjectEntryId());
+
+		for (Iterator<Node> iterator = objectEntryTree.iterator();
+			 iterator.hasNext();) {
```

Comment:
@victorKammerer please use `while`, as it is what we normally use with `iterators` in our code.

```java
Iterator<Node> iterator = objectEntryTree.iterator();

while (iterator.hasNext()) {
```